File: ViewBindingFJava/app/src/main/java/com/websarva/wings/android/viewbindingfjava/ui/MainViewModel.java
package com.websarva.wings.android.viewbindingfjava.ui;

import androidx.lifecycle.ViewModel;

public class MainViewModel extends ViewModel {
	private double _randNum = 0;

	public void generateNewRandNum() {
		_randNum = Math.random();
	}

	public String getRandNumStr() {
		return String.valueOf(_randNum);
	}
}

File: DataBindingBasicKotlin/app/src/main/java/com/websarva/wings/android/databindingbasickotlin/ui/MainViewModel.kt
package com.websarva.wings.android.databindingbasickotlin.ui

import androidx.lifecycle.ViewModel

class MainViewModel : ViewModel() {
	var randNum = 0.0

	fun generateNewRandNum() {
		randNum = Math.random()
	}

	fun getRandNumStr(): String {
		return randNum.toString()
	}
}
